cli/azd/cmd/package.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/azure/azure-dev/cli/azd/pkg/output/ux"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type packageFlags struct {
+	serviceName string
+	global      *internal.GlobalCommandOptions
+	*envFlag
+}
+
+func newPackageFlags(cmd *cobra.Command, global *internal.GlobalCommandOptions) *packageFlags {
+	flags := &packageFlags{
+		global:  global,
+		envFlag: newEnvFlag(cmd, global),
+	}
+
+	flags.Bind(cmd.Flags(), global)
+
+	return flags
+}
+
+func (pf *packageFlags) Bind(local *pflag.FlagSet, global *internal.GlobalCommandOptions) {
+	local.StringVar(
+		&pf.serviceName,
+		"service",
+		"",
+		//nolint:lll
+		"Deploys a specific service (when the string is unspecified, all services that are listed in the "+azdcontext.ProjectFileName+" file are deployed).",
+	)
+	//deprecate:flag hide --service
+	_ = local.MarkHidden("service")
+}
+
+func newPackageCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "package <service>",
+		Short:  "Packages the application's code to be deployed to Azure.",
+		Hidden: true,
+	}
+	cmd.Args = cobra.MaximumNArgs(1)
+	return cmd
+}
+
+type packageAction struct {
+	flags          *packageFlags
+	args           []string
+	projectConfig  *project.ProjectConfig
+	projectManager project.ProjectManager
+	serviceManager project.ServiceManager
+	console        input.Console
+	formatter      output.Formatter
+	writer         io.Writer
+}
+
+func newPackageAction(
+	flags *packageFlags,
+	args []string,
+	projectConfig *project.ProjectConfig,
+	projectManager project.ProjectManager,
+	serviceManager project.ServiceManager,
+	console input.Console,
+	formatter output.Formatter,
+	writer io.Writer,
+) actions.Action {
+	return &packageAction{
+		flags:          flags,
+		args:           args,
+		projectConfig:  projectConfig,
+		projectManager: projectManager,
+		serviceManager: serviceManager,
+		console:        console,
+		formatter:      formatter,
+		writer:         writer,
+	}
+}
+
+type PackageResult struct {
+	Timestamp time.Time                                `json:"timestamp"`
+	Services  map[string]*project.ServicePackageResult `json:"services"`
+}
+
+func (pa *packageAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+	// Command title
+	pa.console.MessageUxItem(ctx, &ux.MessageTitle{
+		Title: "Packaging services (azd package)",
+	})
+
+	targetServiceName := pa.flags.serviceName
+	if len(pa.args) == 1 {
+		targetServiceName = pa.args[0]
+	}
+
+	serviceNameWarningCheck(pa.console, pa.flags.serviceName, "package")
+
+	if targetServiceName != "" && !pa.projectConfig.HasService(targetServiceName) {
+		return nil, fmt.Errorf("service name '%s' doesn't exist", targetServiceName)
+	}
+
+	if err := pa.projectManager.Initialize(ctx, pa.projectConfig); err != nil {
+		return nil, err
+	}
+
+	if err := pa.ensureTools(ctx, targetServiceName); err != nil {
```

Comment:
This change was already put in but can be modified.  The thoughts behind this is that the devs will need all the required external tools at some point to support provision/deploy, so instead of incrementally failing why not just give them the heads up at one time that they are missing the tools.